src/software/simulation/simulator_robot.cpp

```diff
@@ -464,3 +420,45 @@ void SimulatorRobot::runCurrentPrimitive(std::shared_ptr<FirmwareWorld_t> firmwa
     app_primitive_manager_runCurrentPrimitive(primitive_manager.get(),
                                               firmware_world.get());
 }
+
+void SimulatorRobot::applyDribblerForce(PhysicsRobot *physics_robot,
+                                        PhysicsBall *physics_ball)
+{
+    auto robot = physics_robot->getRobotState();
+    auto ball  = physics_ball->getBallState();
+
+    // To dribble, we apply a force towards the center and back of the dribbling area,
+    // closest to the dribbler damper. We vary the magnitude of the force by how far the
+    // ball is from this "dribbling point". This more-or-less acts like a tiny gravity
+    // well that sucks the ball into place, except with more force the further away the
+    // ball is. Once the ball is no longer in the dribbler area this force is not
+    // applied (it is only applied as long as the ball is in the dribbler area).
+
+    Point dribble_point =
+        robot.position() +
+        Vector::createFromAngle(robot.orientation())
+            .normalize(DIST_TO_FRONT_OF_ROBOT_METERS + BALL_MAX_RADIUS_METERS -
+                       PhysicsRobot::DRIBBLER_DEPTH);
+    Vector dribble_force_vector = dribble_point - ball.position();
+    // convert to cm so we operate on a small scale
+    double dist_from_dribble_point_cm =
+        dribble_force_vector.length() * CENTIMETERS_PER_METER;
```

Comment:
Can we add a newline here? Almost thought this was part of the big comment below.

Reply:
done